### PR TITLE
Add DLP docs

### PR DESCRIPTION
**Pull request by kdorosh**
cc @EItanya 
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1500

### CODE REVIEW

File: docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+curl $(glooctl proxy url)/api/pets
+```
+
+You should obtain the following response:
+
+```json
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+Now, let's pretend the animal names (Cat/Dog) are sensitive, identifying information and mask them.
```
Comment (EItanya) [inline docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md]: This sentence is a little confusing to read, not good at grammar personally though so I can't 100% say why

Comment (kdorosh) [reply docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md]: I think the grammar is technically correct, but if it trips people up at all then it should be changed. The example is possibly a bit too contrived, which doesn't help.

Does this sound any better for now?

"Names are often used a personally identifying information, or PII. Lets write our own regex to mask the names returned by the petstorre endpoint (i.e. Cat or Dog)." and then updating the regex appropriately

Comment (kdorosh) [reply docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md]: clarified example here https://github.com/solo-io/gloo/pull/1574/commits/dffe4eae1a3147dfbf63753231712e2c43f661a2 . was able to get other regexes to work but this one won't work for some reason, even though regex parser confirmed desired behavior (besides the quote handling).

The more important flaw is that the regex here didn't match -- does the regex implementation lack some lookahead/look-behind features? Can you link me to the regex implementation that DLP uses?

Comment (kdorosh) [reply docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md]: also, how hard would it be to add group matching to our regex support?

Comment (kdorosh) [reply docs/content/gloo_routing/virtual_services/data_loss_prevention/_index.md]: Opened https://github.com/solo-io/gloo/issues/1591 and https://github.com/solo-io/gloo/issues/1592 to track

Happy to merge these as is, and update docs as we address these issues

**Pull request merged**